src/java.base/share/classes/sun/security/util/DisabledAlgorithmConstraints.java

```diff
@@ -150,12 +150,13 @@ public DisabledAlgorithmConstraints(String propertyName,
     }
 
     /*
-     * This only checks if the algorithm has been completely disabled.  If
-     * there are keysize or other limit, this method allow the algorithm.
+     * This only checks if the algorithm of cipher suite has been completely
+     * disabled.  If there is keysize or other limit, this method allow the algorithm.
      */
     @Override
     public final boolean permits(Set<CryptoPrimitive> primitives,
```

Comment:
If I move TLS cipher suite check from this method to `checkConstraints` then tests fail. In the context of `AlgorithmConstraints` the "algorithm" word means either an actual algorithm like "RSA" or a TLS cipher suite. So in this method we completely disable given TLS cipher suite if it matches the criteria. I just updated the comment to avoid any confusion.